src/core/ext/transport/cronet/transport/cronet_transport.cc

```diff
@@ -1090,22 +1093,28 @@ static enum e_op_result execute_stream_op(struct op_and_state* oas) {
       grpc_slice_buffer write_slice_buffer;
       grpc_slice slice;
       grpc_slice_buffer_init(&write_slice_buffer);
-      if (1 != stream_op->payload->send_message.send_message->Next(
-                   stream_op->payload->send_message.send_message->length(),
-                   nullptr)) {
-        /* Should never reach here */
-        GPR_ASSERT(false);
-      }
-      if (GRPC_ERROR_NONE !=
-          stream_op->payload->send_message.send_message->Pull(&slice)) {
-        /* Should never reach here */
-        GPR_ASSERT(false);
-      }
-      grpc_slice_buffer_add(&write_slice_buffer, slice);
-      if (GPR_UNLIKELY(write_slice_buffer.count != 1)) {
+      do {
```

Comment:
I suspect that even though this works for now, it may not be ideal and might not work in the future if, e.g., the byte stream in `stream_op->payload->send_message.send_message` changed type. 

The reason is that the `ByteStream` interface does not guarantee that there'll always be an empty slice in an empty byte stream. Particularly, the interface [said](https://github.com/grpc/grpc/blob/88af358f9ec9b2a84f82b3f4e795df62c920c29b/src/core/lib/transport/byte_stream.h#L48) the Next() method "should not be called if there is no data left on the stream". That means the `while` loop in the previous revision was actually the right thing to do. We need to take care of the assert in `write_slice_buffer.count < 1` branch instead.

The limitation of `write_slice_buffer.count >= 1` comes from Cronet IIUC, so we probably can't directly eliminate it (or even if we can, it probably goes to another PR). So I would suggest an alternative workaround: if `write_slice_buffer` is empty after the while loop, add a zero length slice to it (replacing the original `write_slice_buffer.count < 1` branch).

Reply:
done. thanks for reviews.

Reply:
i removed "write_slice_buffer.count == 1" assert. The reason is ** create_grpc_frame** always return 5 byte gRPC header (even _write_slice_buffer_ is **empty**)